src/Blocks/BlockPiston.cpp

```diff
@@ -86,28 +73,129 @@ bool cBlockPistonHandler::GetPlacementBlockTypeMeta(
 
 
 
-int cBlockPistonHandler::FirstPassthroughBlock(int a_PistonX, int a_PistonY, int a_PistonZ, NIBBLETYPE pistonmeta, cWorld * a_World)
+Vector3i cBlockPistonHandler::VectorFromMetaData(int a_PistonMeta)
```

Comment:
Yeah, I'll change it. But the function is just a helper and does not modify the cPistonHandler class. So I'll make this function const too.